File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOSportGoalParamModel.kt
//
//  IDOSportGoalParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.Gson
import com.google.gson.GsonBuilder

///
/// Set exercise goal event
import com.google.gson.annotations.SerializedName

open class IDOSportGoalParamModel(sportStep: Int, walkGoalSteps: Int, targetType: Int?) :
    IDOBaseModel {

    /// Number of steps for the exercise goal                        
    @SerializedName("sport_step")
    var sportStep: Int = sportStep

    /// Walk goal steps per hour setting
    @SerializedName("walk_goal_steps")
    var walkGoalSteps: Int = walkGoalSteps

    /// Target type setting
    /// 0: Invalid
    /// 1: Daily Goal 
    /// 2: Weekly Goal
    /// Requires support from the menu getStepDataTypeV2
    @SerializedName("target_type")
    var targetType: Int? = targetType


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOSportGoalParamModel.swift
//
//  IDOSportGoalParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set exercise goal event
@objcMembers
public class IDOSportGoalParamModel: NSObject, IDOBaseModel {
    /// Number of steps for the exercise goal                        
    public var sportStep: Int
    /// Walk goal steps per hour setting                             
    public var walkGoalSteps: Int
    /// Target type setting
    /// 0: Invalid
    /// 1: Daily Goal 
    /// 2: Weekly Goal
    /// Requires support from the menu `getStepDataTypeV2`
    public var targetType: Int
    
    enum CodingKeys: String, CodingKey {
        case sportStep = "sport_step"
        case walkGoalSteps = "walk_goal_steps"
        case targetType = "target_type"
    }
    
    public init(sportStep: Int,walkGoalSteps: Int,targetType: Int = 0) {
        self.sportStep = sportStep
        self.walkGoalSteps = walkGoalSteps
        self.targetType = targetType
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

